Tests/JetworkingTests/ClientTests/ClientTests.swift

```diff
@@ -395,10 +387,212 @@ final class ClientTests: XCTestCase {
+            dispatchPrecondition(condition: .onQueue(DispatchQueue.main))
+
+            switch result {
+            case .failure:
+                break
+
+            case let .success(resultData):
+                print(resultData)
+            }
+
+            XCTAssertNotNil(response)
+            XCTAssertEqual(response?.statusCode, 200)
+            waitForGetExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 5.0, handler: nil)
+    }
+
+    func testCustomDecoderForPost() {
+        let testableDecoder = TestableDecoder()
+        let client = Client(configuration: makeDefaultClientConfiguration(.sync)) { session in
+            session.configuration.timeoutIntervalForRequest = 30
+        }
+
+        let postDecodingCalledExpectation = expectation(description: "Decoder method called for post")
+        let waitForPostExpectation = expectation(description: "Wait for post")
+
+        let body: MockBody = .init(foo1: "bar1", foo2: "bar2")
+
+        testableDecoder.decodeCalled = {
+            postDecodingCalledExpectation.fulfill()
+        }
+
+        client.post(endpoint: Endpoints.post.overrideStandardDecoderWith(testableDecoder), body: body) { response, result in
+            dispatchPrecondition(condition: .onQueue(DispatchQueue.main))
+            switch result {
+            case .failure:
+                break
+
+            case let .success(resultData):
+                print(resultData)
+            }
+
+            XCTAssertNotNil(response)
+            XCTAssertEqual(response?.statusCode, 200)
+            waitForPostExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 5.0, handler: nil)
+    }
+
+    func testCustomDecoderForPut() {
+        let testableDecoder = TestableDecoder()
+        let client = Client(configuration: makeDefaultClientConfiguration(.sync)) { session in
+            session.configuration.timeoutIntervalForRequest = 30
+        }
+
+        let putDecodingCalledExpectation = expectation(description: "Decoder method called for put")
+        let waitForPutExpectation = expectation(description: "Wait for put")
+
+        let body: MockBody = .init(foo1: "bar1", foo2: "bar2")
+        testableDecoder.decodeCalled = {
+            putDecodingCalledExpectation.fulfill()
+        }
+
+        client.put(endpoint: Endpoints.put.overrideStandardDecoderWith(testableDecoder), body: body) { response, result in
+            dispatchPrecondition(condition: .onQueue(DispatchQueue.main))
+            switch result {
+            case .failure:
+                break
+
+            case let .success(resultData):
+                print(resultData)
+            }
+
+            XCTAssertNotNil(response)
+            XCTAssertEqual(response?.statusCode, 200)
+            waitForPutExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 5.0, handler: nil)
+    }
+
+    func testCustomDecoderForPatch() {
```

Comment:
Is it possible that we extract the customDecoderTests into a separate file?

Reply:
I added a new file for the custom decoder tests